File: CS-Examples/07_Conversion/ToPDF.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace ToPDF
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load file from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ToPDF.xlsx");

            // Set the ConverterSetting property to enable fitting sheets to page during PDF conversion
            workbook.ConverterSetting.SheetFitToPage = true;

            // Save the workbook as a PDF file with the name "sample.pdf"
            workbook.SaveToFile("sample.pdf", FileFormat.PDF);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer("sample.pdf");
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }


	}
}

File: VB-Examples/07_Conversion/ToPDF.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace ToPDF
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			' Create a new instance of Workbook
			Using workbook As New Workbook()

				' Load the Excel file from the specified path
				workbook.LoadFromFile("..\..\..\..\..\..\Data\ToPDF.xlsx")

				' Set the ConverterSetting property to fit the sheet to the page when converting to PDF
				workbook.ConverterSetting.SheetFitToPage = True

				' Save the workbook as a PDF file with the name "sample.pdf"
				workbook.SaveToFile("sample.pdf", FileFormat.PDF)
			End Using

			ExcelDocViewer("sample.pdf")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub


	End Class
End Namespace
